File: lib/features/user/domain/entities/user_entity.dart
import 'package:equatable/equatable.dart';

class LoginUserRequestEntity extends Equatable {
  final String? username;
  final String? mdp;

  const LoginUserRequestEntity({
    this.username,
    this.mdp,
  });

  @override
  List<Object?> get props => [
    username,
    mdp,
  ];
}

class LoginUserResponseEntity extends Equatable {
  final String? id;
  final String? nom;
  final String? prenom;
  final String? mail;
  final String? username;

  const LoginUserResponseEntity({
    this.id,
    this.nom,
    this.prenom,
    this.mail,
    this.username,
  });

  @override
  List<Object?> get props => [
    id,
    nom,
    prenom,
    mail,
    username,
  ];
}

class RegisterUserRequestEntity extends Equatable {
  final String? nom;
  final String? prenom;
  final String? mail;
  final String? mdp;
  final String? username;

  const RegisterUserRequestEntity({
    this.nom,
    this.prenom,
    this.mail,
    this.mdp,
    this.username,
  });

  @override
  List<Object?> get props => [
    nom,
    prenom,
    mail,
    mdp,
    username
  ];
}

class RegisterUserResponseEntity extends Equatable {
  final String? id;
  final String? nom;
  final String? prenom;
  final String? mail;
  final String? username;

  const RegisterUserResponseEntity({
    this.id,
    this.nom,
    this.prenom,
    this.mail,
    this.username,
  });

  @override
  List<Object?> get props => [
    id,
    nom,
    prenom,
    mail,
    username,
  ];
}

File: src/users/entities/user.entity.ts
import { Entity, PrimaryGeneratedColumn, Column, Unique, ManyToMany } from "typeorm";
import { Friendslist } from "../../friendslist/entities/friendslist.entity";
import { Filmotheque } from "../../filmotheque/entities/filmotheque.entity";
import { Bibliotheque } from "../../bibliotheque/entities/bibliotheque.entity";

@Entity()
@Unique(["mail"])
@Unique(["username"])
export class User {
  @PrimaryGeneratedColumn('uuid')
  id: string;

  @Column({unique: true})
  username: string;

  @Column()
  nom: string;

  @Column()
  prenom: string;

  @Column({unique: true})
  mail: string;

  @Column({select: false})
  mdp: string;

  @ManyToMany(() => Friendslist, friendslist => friendslist.friends)
  friends: Friendslist[];

  @ManyToMany(() => Filmotheque, filmotheque => filmotheque.users)
  filmotheques: Filmotheque[];

  @ManyToMany(() => Bibliotheque, bibliotheque => bibliotheque.users)
  bibliotheques: Bibliotheque[];
}